grails-app/controllers/org/pih/warehouse/notification/NotificationController.groovy

```diff
@@ -104,63 +90,206 @@ class NotificationController {
+                    JSONObject userMapping = toJson(grailsApplication.config?.openboxes?.integration?.userMapping ?: null)
+                    log.info "UserMapping:${userMapping}"
+                    if(userMapping) {
+                        String requestedByString = oJson.getString("requestedBy")
+                        if(userMapping.has(requestedByString)) {
+                            String username = userMapping.getString(requestedByString)
+                            log.info "RequestedByString:${requestedByString}, userMapping[$requestedByString]:${username}"
+                            requestedBy = User.findByUsername(username)
+                        }
+                    }
                 }
-//            }
+                requisition.requestedBy = requestedBy
+            }
+        } catch (Exception ex) {
+            log.error "Error:${ex.printStackTrace()} in reading requestedBy from order json:${oJson}"
+        }
+        try {
+            if (oJson.has("dateRequested") && oJson.getString("dateRequested")) {
+                Date dateRequested = DateUtils.parseDate(oJson.getString("dateRequested"), "yyyy-MM-dd")
+                requisition.dateRequested = dateRequested
+            } else {
+                requisition.dateRequested = new Date()
+            }
+        } catch (Exception ex) {
+            log.error "Error:${ex.printStackTrace()} in reading dateRequested from order json:${oJson}"
+            requisition.dateRequested = new Date()
+        }
+        try {
+            if (oJson.has("requestedDeliveryDate") && oJson.getString("requestedDeliveryDate")) {
+                Date dateRequested = DateUtils.parseDate(oJson.getString("requestedDeliveryDate"), "yyyy-MM-dd")
+                requisition.dateRequested = dateRequested
+            } else {
+                requisition.requestedDeliveryDate = new Date()
+            }
+        } catch (Exception ex) {
+            log.error "Error in reading dateRequested from order json:${oJson}"
+            requisition.requestedDeliveryDate = new Date()
+        }
+        Boolean lineItemHasError = false
+        JSONArray orderLineItemsArray = oJson.getJSONArray("orderItems")
+        for (int lIndex = 0; lIndex < orderLineItemsArray?.length(); lIndex++) {
+            JSONObject lineItem = orderLineItemsArray.getJSONObject(lIndex)
+            log.info "Reading LineItem:${lineItem} for Order"
             try {
-                if (!productInstance?.validate() || productInstance?.hasErrors()) {
-                    productInstance?.errors?.allErrors?.each {
-                        log.error("ERR:${it}")
+                RequisitionItem requisitionItem = new RequisitionItem()
+                Product product = Product.findByUpc(lineItem.getString("productId"))
+                if (!product) {
+                    JSONObject productMapping = toJson(grailsApplication.config?.openboxes?.integration?.productMapping ?: null)
```

Comment:
Let's remove the product mapping and throw an IllegalArgumentException if we can't find a suitable product given the ID passed by the external system.